pkg/maps/ctmap/ipv4.go

```diff
@@ -0,0 +1,160 @@
+package ctmap
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/cilium/cilium/common"
+	"github.com/cilium/cilium/common/types"
+	"github.com/cilium/cilium/pkg/bpf"
+	"github.com/cilium/cilium/pkg/u8proto"
+	"net"
+	"unsafe"
+)
+
+var (
+	// Service4Map is the map representing the bpf local conntrack map.
+	Service4Map = bpf.NewMap(MapName4+"_services",
+		bpf.MapTypeHash,
+		int(unsafe.Sizeof(CtKey4{})),
+		int(unsafe.Sizeof(CtEntry{})),
+		MapNumEntriesLocal)
+
+	// Service4GlobalMap is the map representing the bpf global conntrack map.
+	Service4GlobalMap = bpf.NewMap(MapName4Global+"services",
+		bpf.MapTypeHash,
+		int(unsafe.Sizeof(CtKey4Global{})),
+		int(unsafe.Sizeof(CtEntry{})),
+		MapNumEntriesGlobal)
+)
+
+//CtKey4 represents the key for IPv4 entries in the local BPF conntrack map.
+type CtKey4 struct {
+	addr    types.IPv4
+	sport   uint16
+	dport   uint16
+	nexthdr u8proto.U8proto
+	flags   uint8
+}
+
+// NewCtKey4 creates a CtKey4 with the provided ip, source port, destination port, next header, and flags.
+func NewCtKey4(addr net.IP, sport uint16, dport uint16, nexthdr u8proto.U8proto, flags uint8) *CtKey4 {
+	key := CtKey4{
+		sport:   sport,
+		dport:   dport,
+		nexthdr: nexthdr,
+		flags:   flags,
+	}
+
+	copy(key.addr[:], addr.To4())
+
+	return &key
+}
+
+// GetKeyPtr returns the unsafe.Pointer for k.
+func (k *CtKey4) GetKeyPtr() unsafe.Pointer { return unsafe.Pointer(k) }
+
+//NewValue creates a new bpf.MapValue.
+func (k *CtKey4) NewValue() bpf.MapValue { return &CtEntry{} }
+
+// Map returns the BPF map for the local IPv4 conntrack map.
+func (k CtKey4) Map() *bpf.Map { return Service4Map }
```

Comment:
lbmap uses `Map()` to hide name for IPv4 and IPv6 specific map names for services map. The difference is that lbmaps are always global, just per address family whereas ctmap is per endpoint unless global.

I think all of the `Map()` usage can be removed and what remains is the address family specific `Convert()`.